torch/csrc/utils/disable_torch_function.cpp

```diff
@@ -143,6 +148,46 @@ PyObject* THPModule_disable_torch_function(PyObject* self, PyObject* a) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPModule_skip_one_hop_torch_function(
+    PyObject* /*self*/,
+    PyObject* a) {
+  HANDLE_TH_ERRORS
+  PyObject *func = nullptr, *types = nullptr, *args = nullptr,
+           *kwargs = nullptr;
+  if (!PyArg_ParseTuple(a, "OOOO", &func, &types, &args, &kwargs)) {
+    return nullptr;
+  }
+  py::tuple py_args;
+  if (args == Py_None) {
+    py_args = py::make_tuple();
+  } else {
+    py_args = py::reinterpret_borrow<py::tuple>(args);
+  }
+
+  // PyObject_Call allows kwargs to be nullptr.
+  if (kwargs == Py_None) {
+    kwargs = nullptr;
+  } else {
+    TORCH_CHECK_TYPE(PyDict_Check(kwargs), "kwargs must be a dictionary");
+  }
+
+  // PyObject_Call is a C-API calls so no exceptions will be raised
+  // and therefore no need for RAII approach to storing the old value.
+  TORCH_CHECK(
+      !at::impl::PythonTorchFunctionTLS::peek_skip_next(),
+      "skip_one_hop_torch_function called but skip_next_torch_function was already true!");
+  at::impl::PythonTorchFunctionTLS::exchange_skip_next(true);
+  auto result = py::reinterpret_steal<py::object>(
+      PyObject_Call(func, py_args.ptr(), kwargs));
+  bool prev_skip = at::impl::PythonTorchFunctionTLS::exchange_skip_next(false);
+  TORCH_CHECK(
+      !prev_skip || !result,
```

Comment:
This doesn't look right. If the inner function call failed, it may have failed before we got to the `has_torch_function` test. So it is best to reset the skip next but then propagate the *original* Python error in this case.

Reply:
If the call fails, then the `!result` check passes and no exception is raised.

Reply:
oh oops, misread